source/hardware/controllers/pioneer_ddj_400.rst

```diff
@@ -0,0 +1,185 @@
+4     EQs (:hwlabel:`LOW` :hwlabel:`MID` :hwlabel:`HIGH`)      Boosts or cuts channel frequencies
+5     Channel level indicators                                 Displays channel levels (before being sent through channel faders)
+6     :hwlabel:`HEADPHONES` :hwlabel:`MIXING`                  Adjusts between headphone :hwlabel:`CUE` and :hwlabel:`MASTER` levels sent to headphones
+7     :hwlabel:`FILTER`                                        Turn left to apply low pass filter to channel, right to apply high pass filter
+8     :hwlabel:`HEADPHONES` :hwlabel:`LEVEL`                   Adjusts output level from :hwlabel:`PHONES` terminal
+9     Headphone :hwlabel:`CUE` (Channel)                       Turn on to output channel to headphones
+9     :hwlabel:`SHIFT` + Headphone :hwlabel:`CUE`              Toggles quantize for the current channel
+10    Channel fader                                            Adjusts channel levels output to master
+11    Crossfader                                               Fade between left and right channels
+====  =======================================================  ======================================================================
+
+Effect section (p. 9)
+~~~~~~~~~~~~~~~~~~~~~
+
+Mixxx's `standard FX mappings <https://github.com/mixxxdj/mixxx/wiki/Standard-Effects-Mapping>`__
+are implemented as much as is possible using Pioneer's :hwlabel:`BEAT FX` controls.
+
+.. figure:: ../../_static/controllers/pioneer_ddj_400_effects.svg
+   :align: center
+   :width: 20%
+   :figwidth: 100%
+   :alt: Pioneer DDJ-400 (effect section)
+   :figclass: pretty-figures
+
+   Pioneer DDJ-400 (effect section)
+
+====  =======================================================  ======================================================================
+No.   Control                                                  Function
+====  =======================================================  ======================================================================
+1     :hwlabel:`BEAT` :hwlabel:`<`                             Focus on effect slot 1. Press again to unfocus (in order to use wet/dry mix as described below)
+1     :hwlabel:`SHIFT` + :hwlabel:`BEAT` :hwlabel:`<`          Select previous effect (on currently focused effect slot)
+2     :hwlabel:`BEAT` :hwlabel:`>`                             Focus on effect slot 2. Press again to unfocus (in order to use wet/dry mix as described below)
+2     :hwlabel:`SHIFT` + :hwlabel:`BEAT` :hwlabel:`>`          Select next effect (on currently focused effect slot)
+3     :hwlabel:`FX SELECT`                                     Focus on effect slot 3. Press again to unfocus (in order to use wet/dry mix as described below)
+4     Channel select switch                                    Send effect chain mix to channel 1, 2 or master
+5     :hwlabel:`LEVEL/DEPTH`                                   Controls the metaknob of the currently focused effect *if the effect is enabled*.
```

Comment:
Why is the metaknob not controlled if the effect is disabled? Does it switch to controlling the mix knob if the effect is disabled?

Reply:
Yes it does switch to the mix knob if disabled. A bit counterintuitive, and I'm sure about why it was implemented this way. 

In my old mappings this worked the other way around. If an effect was disabled, the metaknob was used. If the an effect was enabled, the light blinks and the mix knob is used.  

What do you think? Should this change? Personally I'd prefer it... but didn't want to tread on anyone's code without discussion.

Reply:
Changing that seems reasonable to me. Of course that would prevent adjusting the metaknob while the effect is enabled, but that seems like a useful tradeoff. There's only so much you can do with a single knob.